Sample Applications/ExpenseIt/ExpenseItDemo/LineItemCollection.cs

```diff
@@ -11,12 +11,13 @@ public class LineItemCollection : ObservableCollection<LineItem>
     {
         public event EventHandler LineItemCostChanged;
 
-        public void InitializeItems()
+        public new void InsertItem(int index, LineItem item)
         {
-            for(int i = 0; i < Items.Count; i++)
+            if (item != null)
             {
-                Items[i].PropertyChanged += LineItemPropertyChanged;
+                item.PropertyChanged += LineItemPropertyChanged;
             }
+            base.InsertItem(index, item);
         }
 
         public new void Add(LineItem item)
```

Comment:
Remove this method - otherwise you'll add the PropertyChanged handler twice.

Reply:
I chose to revert the last modification and remain with the EnsureInitialize solution because the behaviour was not correct at all. With this version, athough we use more lines of code, the result seems to be the safest and it solves the bug.

Reply:
What was wrong?  Moving the "add PropertyChanged listener" logic from Add to InsertItem should work, and eliminate the need for both EnsureInitialized and the "new Add" method.  That uses fewer lines, but more importantly it demonstrates the right way to do the job.   (The presence of the 'new' keyword is often a sign that you're doing something fishy.)

All the IList and IList<T> "add" methods - Add, Insert - end up calling InsertItem.  Unless I'm missing something?

Reply:
Okay, I understand. I will try to change this one again to make it in the correct way.

Reply:
> What was wrong? Moving the "add PropertyChanged listener" logic from Add to InsertItem should work, and eliminate the need for both EnsureInitialized and the "new Add" method. That uses fewer lines, but more importantly it demonstrates the right way to do the job. (The presence of the 'new' keyword is often a sign that you're doing something fishy.)
> 
> All the IList and IList "add" methods - Add, Insert - end up calling InsertItem. Unless I'm missing something?

I was mistaken. The the "add PropertyChanged listener" in the InsertItem method didn't works. It seems that the list still doesn't subscribe to the events of the ones in the markup.